### PR TITLE
Update internal code for torch.geqrf

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56257 Fix MAGMA qr for empty batched inputs
* #56256 Add cuSOLVER path for torch.linalg.qr
* #56255 Remove size arguments for internal orgqr and geqrf calls
* #56254 Add non-allocating helper function for torch.linalg.qr
* #56253 Add cuBLAS path for batched torch.geqrf
* #56252 Add cuSOLVER path for torch.geqrf
* #56251 Port CUDA torch.geqrf to ATen
* **#56250 Update internal code for torch.geqrf**
* #56249 Port CPU torch.geqrf to ATen

Moved `apply_geqrf` to `BatchLinearAlgebraKernel.cpp`. Added
`geqrf_stub` dispatch.

Differential Revision: [D27907362](https://our.internmc.facebook.com/intern/diff/D27907362)

### CODE REVIEW

File: aten/src/ATen/native/BatchLinearAlgebraKernel.cpp [hunk 1]
```diff
@@ -330,6 +330,75 @@ void linalg_eigh_kernel(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos
       });
 }
 
+/*
+  The geqrf function computes QR decomposition of matrices stored in `input`.
```
Comment (mruberry) [inline aten/src/ATen/native/BatchLinearAlgebraKernel.cpp:334]: "computes the QR"

File: aten/src/ATen/native/BatchLinearAlgebraKernel.cpp [hunk 2]
```diff
@@ -330,6 +330,75 @@ void linalg_eigh_kernel(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos
...
+  However, rather than producing a Q matrix directly, it produces a sequence of
```
Comment (mruberry) [inline aten/src/ATen/native/BatchLinearAlgebraKernel.cpp:335]: This is a really nice description of the operation

File: aten/src/ATen/native/BatchLinearAlgebraKernel.cpp [hunk 3]
```diff
@@ -330,6 +330,75 @@ void linalg_eigh_kernel(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos
...
+            implicitly defining Q.
+  * `m` - The number of rows of `input` to consider
+  * `n` - The number of columns of `input` to consider (actual sizes of `input` could be larger)
+
+  For further details, please see the LAPACK documentation for GEQRF.
+*/
+template <typename scalar_t>
+static void apply_geqrf(const Tensor& input, const Tensor& tau, int64_t m, int64_t n) {
+#ifndef USE_LAPACK
+  AT_ERROR("geqrf: LAPACK library not found in compilation");
```
Comment (mruberry) [inline aten/src/ATen/native/BatchLinearAlgebraKernel.cpp:356]: TORCH_CHECK and double check this error message for consistency with other LAPACK not found errors, I think we often say something diferent?

**Review by mruberry — APPROVED** [review]
Just two very small nits on this one; nice work!

**Comment by IvanYashchuk** [comment]
@mruberry I updated this PR and https://github.com/pytorch/pytorch/pull/56249, I think they both can be imported now.

**Pull request closed without merging**